Pages/js/options.js

```diff
@@ -85,19 +85,19 @@ function saveWhitelist() {
   // Each domain can be from 1-63 characters and may contain alphanumeric 
   // characters and - but may not begin or end with - Each top level domain may
   // be from 2 to 9 characters and may contain alpha characters
-  var validateSubdomain = new RegExp("^(?!\-|_)[\\w\-]{1,63}","g"); //subdomains
-  var validateDomain = new RegExp("^(?!\-)[a-zA-Z0-9\-?]{1,63}$","g"); //domain
-  var validateTLD = new RegExp("^[a-zA-Z]{2,9}$","g"); //top level domain
+  var validateSubdomain = new RegExp(/^(?!\-|_)[\\w\-]{1,63}/, "g"); //subdomains
+  var validateDomain = new RegExp(/^(?!\-)[a-zA-Z0-9\-?]{1,63}$/, "g"); //domain
+  var validateTLD = new RegExp(/^[a-zA-Z]{2,9}$/, "g"); //top level domain
   
   //needed because js regex does not have look-behind
-  var notEndInHyphenOrUnder = new RegExp("[^\-_]$","g"); 
-  var notEndInHyphen = new RegExp("[^\-]$","g");
+  var notEndInHyphenOrUnder = new RegExp(/[^\-_]$/, "g"); 
+  var notEndInHyphen = new RegExp(/[^\-]$/, "g");
 
   var domain_regexp = "";  //stores regex to match validated domains
   var valid_domains = [];  //stores validated domains
   
   //iterate over entered list, split by invalid chars
-  for (var i = 0; i < domains.length; i++){ 
+  for (i = 0; i < domains.length; i++){ 
```

Comment:
Ohh, I see what you mean: if you don't declare the variable, it is global. However, "var i" was declared earlier in the function (in line 71), so I didn't remove *all* declarations, just a redundant one. It is still local to the function.

Would it be more clear if I moved "var i;" to the top of the function and then used "i = 0" in both for loops? Or is it okay this way?